src/python/espressomd/reaction_ensemble.pyx

```diff
@@ -37,10 +38,18 @@ cdef class reaction_ensemble(object):
                        gradients from the short ranged interaction potential
                        when using the Reaction ensemble together with a MD
                        scheme.
-        
+
     """
-    def __init__(self,*args,**kwargs):
-        self._params=self._default_params()
+    cdef object _params
+    cdef cReactionEnsemble* RE
```

Comment:
Why do you use a pointer here? Can't you just do cdef cReactionEnseble RE, and drop the explicit memory management?

Reply:
Well, if we write it directly like this I get a comiler crash in line 52:
       self.RE = cReactionEnsemble()
                                  ^
------------------------------------------------------------

/tikhome/jlandsgesell/src/espresso/src/python/espressomd/reaction_ensemble.pyx:52:35: Compiler crash in AnalyseExpressionsTransform

ModuleNode.body = StatListNode(reaction_ensemble.pyx:1:0)
StatListNode.stats[2] = CClassDefNode(reaction_ensemble.pyx:12:5,
    as_name = u'ReactionEnsemble',
    class_name = u'ReactionEnsemble',
    doc = u'\n    ...
    module_name = u'',
    visibility = u'private')
CClassDefNode.body = StatListNode(reaction_ensemble.pyx:13:4)
StatListNode.stats[0] = DefNode(reaction_ensemble.pyx:46:4,
    modifiers = [...]/0,
    name = u'__init__',
    num_required_args = 1,
    py_wrapper_required = True,
    reqd_kw_flags_cname = '0',
    used = True)
File 'Nodes.py', line 430, in analyse_expressions: StatListNode(reaction_ensemble.pyx:48:8)
File 'Nodes.py', line 4775, in analyse_expressions: SingleAssignmentNode(reaction_ensemble.pyx:52:35)
File 'Nodes.py', line 4887, in analyse_types: SingleAssignmentNode(reaction_ensemble.pyx:52:35)

Compiler crash traceback from this point on:
  File "/usr/lib/python2.7/dist-packages/Cython/Compiler/ExprNodes.py", line 4568, in analyse_as_type_constructor
    self.function = RawCNameExprNode(self.function.pos, constructor.type)
AttributeError: 'NoneType' object has no attribute 'type'
src/python/espressomd/CMakeFiles/reaction_ensemble.dir/build.make:92: recipe for target 'src/python/espressomd/reaction_ensemble.cpp' failed
make[2]: *** [src/python/espressomd/reaction_ensemble.cpp] Error 1
make[2]: *** Deleting file 'src/python/espressomd/reaction_ensemble.cpp'
CMakeFiles/Makefile2:4246: recipe for target 'src/python/espressomd/CMakeFiles/reaction_ensemble.dir/all' failed
make[1]: *** [src/python/espressomd/CMakeFiles/reaction_ensemble.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2

Reply:
It looks to me like this reported bug in cython: https://github.com/cython/cython/issues/1791